exp/api/v1beta1/ocimachinepool_machine_types.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright (c) 2022 Oracle and/or its affiliates.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+// +kubebuilder:object:generate=true
+// +groupName=infrastructure.cluster.x-k8s.io
+
+const (
+	Managed     MachineTypeEnum = "MANAGED_TYPE"
+	Virtual     MachineTypeEnum = "VIRTUAL_TYPE"
+	SelfManaged MachineTypeEnum = "SELF_MANAGED_TYPE"
+)
+
+type MachineTypeEnum string
+
+// OCIMachinePoolMachineSpec defines the desired state of OCIMachinePoolMachine
+type OCIMachinePoolMachineSpec struct {
+	// ProviderID is the OCID of the associated machine..
+	// +optional
+	ProviderID *string `json:"providerID,omitempty"`
+
+	// OCID is the OCID of the associated instance.
```

Comment:
NIT: maybe `// OCID is the Oracle Cloud Identifier of the associated instance.` or something. Just instead of using OCID to define OCID 😊